### PR TITLE
Fix proto port json def

### PR DESCRIPTION
**Pull request by robbrockbank**
Set the json tags for the ProtoPort fields to ensure capitalization is correct.

### CODE REVIEW

File: lib/apis/v2/felixconfig.go [hunk 1]
```diff
@@ -187,8 +187,8 @@ type FelixConfigurationSpec struct {
 }
 
 type ProtoPort struct {
-	Protocol string
-	Port     uint16
+	Protocol string `json:"protocol"`
+	Port     uint16 `json:"port"`
```
Comment (matthewdupre) [inline lib/apis/v2/felixconfig.go]: How does this work if people don't want to specify the port, just the protocol?

Comment (caseydavenport) [reply lib/apis/v2/felixconfig.go]: I think we want `omitempty`

Comment (robbrockbank) [reply lib/apis/v2/felixconfig.go]: oh ok - didn't realize that was a thing.  okey dokey - i'll update.

Comment (robbrockbank) [reply lib/apis/v2/felixconfig.go]: i assume that's just for ports.

do we need to be able to specify ranges of ports or multiple ports in the same thing or just put this back as a string that we'll have to validate or???

Comment (robbrockbank) [reply lib/apis/v2/felixconfig.go]: @matthewdupre @caseydavenport ^^^

**Pull request closed without merging**